React API Reference translation (#39)
reactjs/hu.reactjs.org: content/docs/reference-react.md

 - [`React.Component`](#reactcomponent)
 - [`React.PureComponent`](#reactpurecomponent)

Ha nem használsz ES6 osztályokat, használhatod a `create-react-class` modult. Több információért lásd: [A React használata ES6 nélkül](/docs/react-without-es6.html) fejezetet.

A React komponensek függvényekként is definiálhatóak, amiket be is tudunk csomagolni:

- [`React.memo`](#reactmemo)

### React elemek készítése {#creating-react-elements}

A kezelőfelületed leírásához a [JSX használatát](/docs/introducing-jsx.html) ajánljuk. Minden JSX elem csak szintaktikus cukor a [`React.createElement()`](#createelement) meghívásához. Ha JSX-t használsz, a következő metódusokat nem fogod közvetlenül meghívni.

- [`createElement()`](#createelement)
- [`createFactory()`](#createfactory)

Több információért lásd a [React JSX nélkül](/docs/react-without-jsx.html) fejezetet.

### Elemek transzformálása {#transforming-elements}

A `React` több API-t is kínál elemek manipulálásához:

- [`cloneElement()`](#cloneelement)
- [`isValidElement()`](#isvalidelement)
- [`React.Children`](#reactchildren)

### Töredékek {#fragments}

> Ahhoz, hogy több elemet tudj egyszerre renderelni anélkül hogy azokat becsomagolnád egy másik komponensbe, a `React` egy saját komponenst szolgáltat.

Reviewer: ```suggestion
Ahhoz, hogy több elemet tudj egyszerre renderelni anélkül, hogy azokat becsomagolnád egy másik komponensbe, a `React` egy saját komponenst szolgáltat.
```

* * *

## Referencia {#reference}

### `React.Component` {#reactcomponent}

A `React.Component` a React komponensek alaposztálya abban az esetben, ha azok [ES6 osztályokkal](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Classes) vannak definiálva:

```javascript
class Greeting extends React.Component {
  render() {
    return <h1>Helló, {this.props.name}</h1>;
  }
}
```

Ha egy listát szeretnél a `React.Component` osztály metódusairól és tulajdonságairól, nézd meg a [React.Component API referenciát](/docs/react-component.html).

* * *

### `React.PureComponent` {#reactpurecomponent}

A `React.PureComponent` hasonló a [`React.Component`](#reactcomponent)-hez. A különbség annyi, hogy a [`React.Component`](#reactcomponent) nem implementálja a [`shouldComponentUpdate()`](/docs/react-component.html#shouldcomponentupdate) metódust, míg a `React.PureComponent` igen, egy sekély prop és állapot összehasonlítással.

Ha a React komponensed `render()` függvénye ugyanazt az eredményt rendereli ugyanazon propok és állapot esetében, akkor néhány esetben használhatod a `React.PureComponent`-t a teljesítmény fokozása érdekében.

> Megjegyzés
>

> > A `React.PureComponent` `shouldComponentUpdate()` metódusa csak sekély objektum összehasonlítást végez. Ha az objektumok komplex adatstruktúrákat tartalmaznak, az hamisan negatívat eredményezhet mélyebb különbségek esetében. Csak akkor terjessz ki a `PureComponent`-el, ha egyszerű propokra és állapotra számítasz, vagy használd a [`forceUpdate()`](/docs/react-component.html#forceupdate) metódust ha tudod hogy a mély adatstruktúrák megváltoztak. Vagy vedd fontolóra [megváltoztathatatlan objektumok](https://facebook.github.io/immutable-js/) használatát a mélyebb adatstruktúrák gyors összehasonlításának megkönnyítése érdekében.

Reviewer: ```suggestion
> A `React.PureComponent` `shouldComponentUpdate()` metódusa csak sekély objektum összehasonlítást végez. Ha az objektumok komplex adatstruktúrákat tartalmaznak, az hamisan negatívat eredményezhet mélyebb különbségek esetében. Csak akkor terjessz ki a `PureComponent`-el, ha egyszerű propokra és állapotra számítasz, vagy használd a [`forceUpdate()`](/docs/react-component.html#forceupdate) metódust ha tudod, hogy a mély adatstruktúrák megváltoztak. Vagy vedd fontolóra [megváltoztathatatlan objektumok](https://facebook.github.io/immutable-js/) használatát a mélyebb adatstruktúrák gyors összehasonlításának megkönnyítése érdekében.
```

)
```

Egy adott típusú [React elemet](/docs/rendering-elements.html) készít és ad vissza. A type argumentum lehet egy címke név sztring (mint például `'div'` vagy `'span'`), egy [React komponens](/docs/components-and-props.html) típus (akár osztály vagy függvény), vagy egy [React töredék](#reactfragment) típus.

> A [JSX](/docs/introducing-jsx.html)-ben írt kód át lesz konvertálva, hogy az a `React.createElement()`-et használja. Ha JSX-et használsz, a `React.createElement()`-et tipikusan nem kell közvetlenül meghívnod. Nézd meg a [React JSX nélkül](/docs/react-without-jsx.html) fejezetet ha többet akarsz megtudni.

Reviewer: ```suggestion
A [JSX](/docs/introducing-jsx.html)-ben írt kód át lesz konvertálva, hogy az a `React.createElement()`-et használja. Ha JSX-et használsz, a `React.createElement()`-et tipikusan nem kell közvetlenül meghívnod. Nézd meg a [React JSX nélkül](/docs/react-without-jsx.html) fejezetet, ha többet akarsz megtudni.
```

)
```

Egy `element`-et alapul véve egy új React elemet klónoz és ad vissza. A keletkezett elem rendelkezni fog az eredeti elem és az új propok sekély összefonásával. Az új gyermekek átveszik a meglévő gyermekek helyét. A `key` és `ref` attribútumok meg lesznek tartva az eredeti elemből.

A `React.cloneElement()` majdnem ekvivalens ezzel:

```js
<element.type {...element.props} {...props}>{children}</element.type>
```

> De a `ref` attribútumokat is megőrzi. Ez azt jelenti, hogy ha egy olyan gyermeket kapsz ami rendelkezik `ref`-el, akkor azt nem fogod véletlenül sem ellopni az ősöktől. Az új elemhez ugyanaz a `ref` lesz hozzákapcsolva.

Reviewer: ```suggestion
De a `ref` attribútumokat is megőrzi. Ez azt jelenti, hogy ha egy olyan gyermeket kapsz, ami rendelkezik `ref`-el, akkor azt nem fogod véletlenül sem ellopni az ősöktől. Az új elemhez ugyanaz a `ref` lesz hozzákapcsolva.
```

React.createFactory(type)
```

Egy függvényt ad vissza ami bizonyos típusú React elemeket produkál. Mint ahogy a [`React.createElement()`](#createelement) esetében is, a type argumentum lehet egy címke név sztring (mint például `'div'` vagy `'span'`), egy [React komponens](/docs/components-and-props.html) típus (akár osztály vagy függvény), vagy egy [React töredék](#reactfragment) típus.

> Ez a segédfüggvény egy korábbról örökölt függvénynek számít, és arra biztatunk hogy inkább használj JSX-et, vagy közvetlenül a `React.createElement()`-et.

Reviewer: ```suggestion
Ez a segédfüggvény egy korábbról örökölt függvénynek számít, és arra biztatunk, hogy inkább használj JSX-et, vagy közvetlenül a `React.createElement()`-et.
```

> Ha JSX-et használsz, a `React.createFactory()`-t általában nem fogod közvetlenül meghívni. Nézd meg a [React JSX nélkül](/docs/react-without-jsx.html) fejezetet ha többet akarsz megtudni.

Reviewer: ```suggestion
Ha JSX-et használsz, a `React.createFactory()`-t általában nem fogod közvetlenül meghívni. Nézd meg a [React JSX nélkül](/docs/react-without-jsx.html) fejezetet, ha többet akarsz megtudni.
```

React.isValidElement(object)
```

Azt ellenőrzi, hogy az objektum érvényes React elemnek minősül-e. `true` vagy `false` értéket ad vissza.

* * *

### `React.Children` {#reactchildren}

> A `React.Children` segédeszközként szolgál ha a nem áttetsző `this.props.children` adatstruktúrával kell dolgozni.

Reviewer: ```suggestion
A `React.Children` segédeszközként szolgál, ha a nem áttetsző `this.props.children` adatstruktúrával kell dolgozni.
```

```

A nem áttetsző `children` adatstruktúrát adja vissza egy lapos tömbként, egy kulcsot rendelve minden gyermekhez. Hasznos lehet ha gyermekek listáját akarod manipulálni a render metódusodban, különösen ha át akarod rendezni, vagy le akarsz vágni a `this.props.children`-ből mielőtt azt lejjebb adod.

> Megjegyzés:
>
> Gyermekek listájának lapítása közben a `React.Children.toArray()` megváltoztatja a kulcsokat a beágyazott tömbök szemantikájának megtartása érdekében. Azaz a `toArray` a visszaadott tömbben minden kulcshoz hozzáad egy prefixumot, hogy minden elem kulcsa az azt tartalmazó tömb hatókörében legyen.

* * *

### `React.Fragment` {#reactfragment}

A `React.Fragment` komponens több elem visszaadását teszi lehetővé a `render()` metódusban anélkül, hogy új DOM elemet hozna létre.

```javascript
render() {
  return (
    <React.Fragment>
      Valami szöveg.
      <h2>Egy fejléc</h2>
    </React.Fragment>
  );
}
```

Használhatod a gyorsított `<></>` szintaxissal is. Még több információért nézd meg a [React v16.2.0 Továbbfejlesztett Töredékek támogatás](/blog/2017/11/28/react-v16.2.0-fragment-support.html) blog posztot.

### `React.createRef` {#reactcreateref}

> A `React.createRef` egy [ref](/docs/refs-and-the-dom.html)-et hoz létre amit a ref attribútummal csatolhatunk React elemekhez.

Reviewer: ```suggestion
A `React.createRef` egy [ref](/docs/refs-and-the-dom.html)-et hoz létre, amit a ref attribútummal csatolhatunk React elemekhez.
```

`embed:16-3-release-blog-post/create-ref-example.js`

### `React.forwardRef` {#reactforwardref}

> A `React.forwardRef` egy React komponenst hoz létre ami továbbadja a [ref](/docs/refs-and-the-dom.html) attribútumot amit kap egy másik komponensnek lejjebb a komponensfában. Ezt nem gyakran kell alkalmazni, de két speciális esetben hasznos tud lenni:

Reviewer: ```suggestion
A `React.forwardRef` egy React komponenst hoz létre, ami továbbadja a kapott [ref](/docs/refs-and-the-dom.html) attribútumot egy másik komponensnek lejjebb a komponensfában. Ezt nem gyakran kell alkalmazni, de két speciális esetben hasznos tud lenni:
```

* [Refek továbbítása DOM elemeknek](/docs/forwarding-refs.html#forwarding-refs-to-dom-components)
* [Refek továbbítása felsőbb rendű komponenseknek](/docs/forwarding-refs.html#forwarding-refs-in-higher-order-components)

> A `React.forwardRef` egy renderelő függvényt fogad argumentumként. A React ezt a függvényt hívja meg a `props` és `ref` argumentumokkal. Ez a függvény egy React csomópontot kell hogy visszaadjon.

Reviewer: ```suggestion
A `React.forwardRef` egy renderelő függvényt fogad argumentumként. A React ezt a függvényt hívja meg a `props` és `ref` argumentumokkal. Ez a függvény egy React csomópontot kell, hogy visszaadjon.
```

`embed:reference-react-forward-ref.js`

A fenti példában a React továbbad egy a `<FancyButton ref={ref}>` elemnek adott `ref`-et második argumentumként a renderelő függvénynek a `React.forwardRef` meghívásában. Ez a renderelő függvény továbbadja a `ref`-et a `<button ref={ref}>` elemnek.

Ennek eredményeképp, azután hogy a React hozzácsatolja a refet, a `ref.current` közvetlenül a `<button>` DOM elem példányára fog mutatni.

További információért nézd meg a [refek továbbítása](/docs/forwarding-refs.html) fejezetet.

### `React.lazy` {#reactlazy}

> A `React.lazy()` segítségével egy dinamikusan betöltődő komponenst tudsz definiálni. Ez segít csökkenteni az összecsomagolt kód méretét úgy, hogy a kezdetleges renderelés által nem használt komponensek betöltése késleltetve lesznek.

Reviewer: ```suggestion
A `React.lazy()` segítségével egy dinamikusan betöltődő komponenst tudsz definiálni. Ez segít csökkenteni az összecsomagolt kód méretét úgy, hogy a kezdetleges renderelés által nem használt komponensek betöltése késleltetve lesz.
```

A használatáról többet tanulhatsz a [kód felvágás dokumentációban](/docs/code-splitting.html#reactlazy). Valószínűleg [ezt a cikket](https://medium.com/@pomber/lazy-loading-and-preloading-components-in-react-16-6-804de091c82d) is megnézheted, ami részletesebben is elmagyarázza a használatot.

```js
// Ez a komponens dinamikusan van betöltve
const SomeComponent = React.lazy(() => import('./SomeComponent'));
```

> Jegyezd meg, hogy a `lazy` komponensek renderelése megköveteli, hogy valahol feljebb a komponensfában legyen egy `<React.Suspense>`. Így tudsz megadni egy betöltés indikátort.

Reviewer: ```suggestion
Megjegyzendő, hogy a `lazy` komponensek renderelése megköveteli, hogy valahol feljebb a komponensfában legyen egy `<React.Suspense>`. Így tudsz megadni egy betöltés indikátort.
```

> **Megjegyzés**
>

> > A `React.lazy` használata dinamikus import segítségével megköveteli hogy a Promise objektum elérhető legyen a JS környezetben. Ez IE11 és az alatt egy polyfill használatát követeli meg.

Reviewer: ```suggestion
> A `React.lazy` használata dinamikus import segítségével megköveteli, hogy a Promise objektum elérhető legyen a JS környezetben. Ez IE11 és az alatt egy polyfill használatát követeli meg.
```
